imap_processing/tests/hi/test_hi_l1b.py

```diff
@@ -29,3 +37,117 @@ def test_hi_l1b_de(create_de_data, tmp_path):
     l1b_dataset = hi_l1b(processed_data[0], data_version=data_version)
     assert l1b_dataset.attrs["Logical_source"] == "imap_hi_l1b_45sensor-de"
     assert len(l1b_dataset.data_vars) == 14
+
+
+@pytest.fixture()
+def synthetic_trigger_id_and_tof_data():
+    """Create synthetic minimum dataset for testing the
+    coincidence_type_and_time_deltas algorithm."""
+    # The following coincidence type table shows possible values to consider
+    # Value| # Exp | Requirements to get this value
+    # -----|-------|-------------------------------
+    #   0  |   0   | Non-event not recorded
+    #   1  |   0   | Can't trigger c2 only
+    #   2  |   2   | trigger_id = 3, tof_3 invalid
+    #   3  |   2   | trigger_id = 3, tof_3 valid
+    #   4  |   2   | trigger_id = 2, no valid tofs
+    #   5  |   0   | B and C2 not possible?
+    #   6  |   4   | trigger_id = 2 OR 3, tof_2 valid
+    #   7  |   4   | trigger_id = 2 OR 3, tof_2/3 valid
+    #   8  |   2   | trigger_id = 3, no valid tofs
+    #   9  |   0   | A and C2 not possible?
+    #  10  |   3   | trigger_id = 1, tof_2 OR trigger_id = 3, tof_1
+    #  11  |   3   | trigger_id = 1, tof_2/3, OR trigger_id = 3, tof_1/3
+    #  12  |   2   | trigger_id = 1 OR 2, tof_1
+    #  13  |   0   | A/B and C2 not possible?
+    #  14  |   3   | trigger_id = 1 OR 2 OR 3, tof_1/2
+    #  15  |   3   | trigger_id = 1, 2, 3, tof_1/2/3
+
+    # Use meshgrid to get all combinations of trigger_id and tof valid/invalid
+    # Note: this generates 6 impossible occurrences where C1 is not triggered
+    #    but C2 is. Those are manually removed below.
+    ids = np.arange(3) + 1
+    tof1s = np.array(np.concatenate((HiConstants.TOF1_BAD_VALUES, [1])))
+    tof2s = np.array(np.concatenate((HiConstants.TOF2_BAD_VALUES, [2])))
+    tof3s = np.array(np.concatenate((HiConstants.TOF3_BAD_VALUES, [3])))
+    var_names = ["trigger_id", "tof_1", "tof_2", "tof_3"]
+    data = np.meshgrid(ids, tof1s, tof2s, tof3s)
```

Comment:
Great idea to more easily cover the combinations!

Reply:
I can't quite take credit... found it on SO.